### PR TITLE
[oneDNN ] conv2d fwd&bwd optimization

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Performance optimization

### PR changes
OPs

### Describe
Enables using efficient memory arrangement for training of convolutional models using oneDNN kernels

### CODE REVIEW

**Comment by jczaja** [comment]
@arlesniak Please review

**Comment by jczaja** [comment]
@luotao1 could you please advice on what is the problem with PR-CI-Windows-OPENBLAS (I have triggerred rebuild on previous day)  ?

**Comment by luotao1** [comment]
How about rerun it again?

**Comment by lidanqing-vv** [comment]
@wojtuss  Please help review this

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc [hunk 1]
```diff
@@ -215,14 +215,7 @@ class ConvMKLDNNHandlerT
...
-      // fetch op following it , so numercial grad is computed (in python)
-      // using block formats which will give wrong results
-      const std::string data_format = ctx.Attr<std::string>("data_format");
-      auto chosen_memory_format =
-          is_test ? MKLDNNMemoryFormat::any
-                  : platform::data_format_to_memory_format(data_format);
+      auto chosen_memory_format = MKLDNNMemoryFormat::any;
 
       // Check the format for user's special output
       if (chosen_memory_format != MKLDNNMemoryFormat::any) {
```
Comment (wozna) [inline paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: Is it needed, since we set `chosen_memory_format` earlier?

**Review by wozna — DISMISSED** [review]
Everything with a residual connection looks good for me.

**Review by arlesniak — DISMISSED** [review]
LGTM after fix for Asia comment

File: paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc]: good catch! I deleted dead section of code

**Review by arlesniak — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Pull request merged**